### PR TITLE
add function to check if number series is in date order

### PR DESCRIPTION
**Pull request by VolodySyn**
#### Summary 
Add function to check if number series is in date order

#### Work Item(s) 
Fixes 
[AB#523824](https://dynamicssmb2.visualstudio.com/1fcb79e7-ab07-432a-a3c6-6cf5a88ba4a5/_workitems/edit/523824)

### CODE REVIEW

File: src/Business Foundation/App/NoSeries/src/Single/NoSeriesImpl.Codeunit.al [hunk 1]
```diff
@@ -429,6 +429,13 @@ codeunit 304 "No. Series - Impl."
         exit(OriginalNoSeriesCode);
     end;
 
+    procedure IsNoSeriesInDateOrder(NoSeriesCode: Code[20]): Boolean
+    var
+        NoSeries: Record "No. Series";
+    begin
+        exit(NoSeries.Get(NoSeriesCode) and NoSeries."Date Order")
```
Comment (haoranpb) [inline src/Business Foundation/App/NoSeries/src/Single/NoSeriesImpl.Codeunit.al]: I'm not sure this is universal.

Shouldn't it throw an error if not found?

Comment (VolodySyn) [reply src/Business Foundation/App/NoSeries/src/Single/NoSeriesImpl.Codeunit.al]: implemented in a similar way to:
procedure IsAutomatic(NoSeriesCode: Code[20]): Boolean
    var
        NoSeries: Record "No. Series";
    begin
        if not NoSeries.Get(NoSeriesCode) then
            exit(false);
        exit(NoSeries."Default Nos.");
    end;

File: src/Business Foundation/App/NoSeries/src/Single/NoSeries.Codeunit.al [hunk 2]
```diff
@@ -326,6 +326,18 @@ codeunit 310 "No. Series"
         exit(NoSeriesImpl.GetNoSeriesLine(NoSeriesLine, NoSeriesCode, UsageDate, HideErrorsAndWarnings));
     end;
 
+    /// <summary>
+    /// Use this method to check if number series is marked for date order.
```
Comment (grobyns) [inline src/Business Foundation/App/NoSeries/src/Single/NoSeries.Codeunit.al]:
```suggestion
    /// Returns true if the number series has Date Order set.
```

File: src/Business Foundation/App/NoSeries/src/Single/NoSeries.Codeunit.al [hunk 3]
```diff
@@ -326,6 +326,18 @@ codeunit 310 "No. Series"
...
+    /// </summary>
+    /// <param name="NoSeriesCode">The No. Series code to check.</param>
+    /// <returns>True if the number series is marked for date order, false otherwise.</returns>
+    procedure IsNoSeriesInDateOrder(NoSeriesCode: Code[20]): Boolean
```
Comment (grobyns) [inline src/Business Foundation/App/NoSeries/src/Single/NoSeries.Codeunit.al:335]:
```suggestion
    /// <Remarks>If Date Order is true, numbers must be retrieved in chronological order.</Remarks>
    procedure IsNoSeriesInDateOrder(NoSeriesCode: Code[20]): Boolean
```

Comment (grobyns) [inline src/Business Foundation/App/NoSeries/src/Single/NoSeries.Codeunit.al:335]: I'd add an overload that takes in the No. Series record as well. If you already have the record there is no need to Get it again.

File: src/Business Foundation/App/NoSeries/src/Single/NoSeriesImpl.Codeunit.al (hunk 1 above)
Comment (grobyns) [inline src/Business Foundation/App/NoSeries/src/Single/NoSeriesImpl.Codeunit.al]: I like the more explicit code of IsAutomatic for example. The result is the same but the code is easier to read...

**Review by grobyns — APPROVED** [review]

**Review by grobyns — APPROVED** [review]

**Review by haoranpb — APPROVED** [review]

**Pull request merged**